source/app/dumping.cpp

```diff
@@ -56,6 +57,48 @@ bool callback_scanBuffer(uint64_t& totalBytes, uint64_t bufferSize) {
     return true;
 }
 
+bool readFile(const std::string& path, std::vector<uint8_t>& data) {
+    data.clear();
+
+    struct stat fileStat {};
+    if (lstat(path.c_str(), &fileStat) == -1 || !S_ISREG(fileStat.st_mode)) {
+        std::wstring errorMessage;
+        errorMessage += L"Failed to retrieve info from source file!\n";
+        errorMessage += L"\nDetails:\n";
+        errorMessage += L"Error " + std::to_wstring(errno) + L" when getting stats for:\n";
+        errorMessage += toWstring(path);
+        setErrorPrompt(errorMessage);
+        return false;
+    }
+
+    int handle = open(path.c_str(), O_RDONLY);
+    if (handle == -1) {
+        std::wstring errorMessage;
+        errorMessage += L"Couldn't open the file to read from!\n";
+        errorMessage += L"\nDetails:\n";
+        errorMessage += L"Error " + std::to_wstring(errno) + L" after opening file!\n";
+        errorMessage += toWstring(path);
+        setErrorPrompt(errorMessage);
+        return false;
+    }
+
+    data.resize(fileStat.st_size);
+    ssize_t bytesRead = read(handle, data.data(), fileStat.st_size);
+    if (bytesRead == -1) {
+        close(handle);
+        std::wstring errorMessage;
+        errorMessage += L"Failed to read all data from this file!\n";
+        errorMessage += L"Error " + std::to_wstring(errno) + L" when reading data from:\n";
+        errorMessage += toWstring(path);
+        setErrorPrompt(errorMessage);
+        return false;
+    }
+
+    close(handle);
+
+    return true;
+}
+
```

Comment:
I think you should use std::ifstream in the relevant spotpass code. I agree with your choice to not try to extend the callback infrastructure to support reading back source files to a buffer or to make a new interface to upload files, but this entire function looks out of place and pretty sizeable for the usage of bypassing the callback infrastructure.

I do appreciate the error handling!

Reply:
Okay